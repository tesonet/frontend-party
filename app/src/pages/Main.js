import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import Redirect from 'react-router/Redirect';
import Header from '../components/Header';
import List from '../components/List';
import * as serversActions from '../actions/servers';
import * as authenticationActions from '../actions/authentication';
import './Main.scss';

export class Main extends Component {
  static propTypes = {
    authenticated: PropTypes.bool.isRequired,
    servers: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      distance: PropTypes.number.isRequired,
    })),
    fetchServers: PropTypes.func.isRequired,
    signoutUser: PropTypes.func.isRequired,
  };

  componentDidMount() {
    this.props.fetchServers();
  }

  onLogout = () => {
    this.props.signoutUser();
  }

  render() {
    const { authenticated, servers } = this.props;
    return !authenticated
    ? <Redirect to="/login" />
    : (
      <div className="page main">
        <Header
          onButtonClick={this.onLogout}
        />
        <List data={servers} />
      </div>
    );
  }
}

const sortByDistanceThenByName = (a, b) => (a.distance === b.distance
  ? a.name.toUpperCase().localeCompare(b.name.toUpperCase())
  : a.distance - b.distance);

export default connect(({ authentication, servers }) => ({
  authenticated: authentication.authenticated,
  servers: servers.sort(sortByDistanceThenByName),
}), { ...serversActions, ...authenticationActions })(Main);
