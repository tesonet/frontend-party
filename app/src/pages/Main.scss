.page.main {
  flex: 1;
  display: flex;
  flex-flow: column;
}
