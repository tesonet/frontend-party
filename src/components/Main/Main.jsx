import React from 'react';

import { CommonContainer } from '@Common';

import { useUserAuthentication } from '../../hooks';
import { LOGGED_IN_MESSAGE, LOGGED_OUT_MESSAGE } from './config/constants';

const Main = () => {
  const { isAuthenticated } = useUserAuthentication();

  return (
    <CommonContainer>
      {isAuthenticated ? LOGGED_IN_MESSAGE : LOGGED_OUT_MESSAGE}
    </CommonContainer>
  );
};
export default Main;
