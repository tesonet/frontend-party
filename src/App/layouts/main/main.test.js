import React from 'react';
import renderer from 'react-test-renderer';
import Main from './main';

it('renders correctly', () => {
  const MainPage = renderer
    .create(<Main />)
    .toJSON();
  expect(MainPage).toMatchSnapshot();
});