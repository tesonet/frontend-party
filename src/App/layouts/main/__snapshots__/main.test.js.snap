// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="sc-dnqmqq frjJcZ"
  >
    <div
      className="sc-EHOje kMbdDN"
    >
      <div
        className="sc-bwzfXH govhE"
        size="small"
      >
        testio
        <div
          className="sc-htpNat iJZniu"
          size="small"
        />
      </div>
      <button
        className="sc-bZQynM kRYBZj btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Logout 
        <span
          className="glyphicon glyphicon-log-out"
        />
      </button>
    </div>
    <div
      className="sc-htoDjs eYwOGw"
    >
      <span>
        SERVER
      </span>
      <span>
        DISTANCE
      </span>
    </div>
  </div>
  <ul
    className="sc-gzVnrw gynILq"
  />
</div>
`;
